src/GitHubExtension/DeveloperId/DeveloperIdProvider.cs

```diff
@@ -102,6 +103,39 @@ public IAsyncOperation<IDeveloperId> LoginNewDeveloperIdAsync()
         }).AsAsyncOperation();
     }
 
+    private DeveloperId LoginNewDeveloperIdWithPAT(Uri hostAddress, SecureString personalAccessToken)
+    {
+        try
+        {
+            GitHubClient gitHubClient = new (new ProductHeaderValue(Constants.DEV_HOME_APPLICATION_NAME), hostAddress);
+            var credentials = new Credentials(new System.Net.NetworkCredential(string.Empty, personalAccessToken).Password);
+            gitHubClient.Credentials = credentials;
+            var newUser = gitHubClient.User.Current().Result;
+            DeveloperId developerId = new (newUser.Login, newUser.Name, newUser.Email, newUser.Url, gitHubClient);
+            SaveOrOverwriteDeveloperId(developerId, personalAccessToken);
+
+            Log.Logger()?.ReportInfo($"{developerId.LoginId} logged in with PAT flow to {developerId.GetHostAddress()}");
+
+            if (developerId is null)

```

Comment:
We check for null here but in `SaveOrOverwriteDeveloperId` we don't check for null. Is there a reason why? Just trying to understand in which circumstances `developerId ` could be null at this point in the function call.  I'm wondering really if the  `DeveloperIds ` list could ever add a `developerId ` thats null.

Reply:
You're right, I can remove this check since this will not be null at this point. This is some leftover code from how the flow was previously. Will fix.

Reply:
Actually, I already fixed this in https://github.com/microsoft/devhomegithubextension/pull/271/commits/ec2c99436fe60e494bd78f23bc091364120e5722

Reply:
ah I see, I must have been looking at the current file while also reviewing this. But sounds good.